app/src/main/java/org/schabi/newpipe/fragments/list/channel/ChannelFragment.java

```diff
@@ -417,9 +431,24 @@ public void handleResult(@NonNull final ChannelInfo result) {
 
         playlistCtrl.setVisibility(View.VISIBLE);
 
-        if (!result.getErrors().isEmpty()) {
-            showSnackBarError(result.getErrors(), UserAction.REQUESTED_CHANNEL,
-                    NewPipe.getNameOfService(result.getServiceId()), result.getUrl(), 0);
+        List<Throwable> errors = result.getErrors();
+        if (!errors.isEmpty()) {
+
+            // handling ContentNotSupportedException not to show the error but an appropriate string
+            // so that crashes won't be sent uselessly and the user will understand what happened
+            Iterator<Throwable> it = errors.iterator();
+            while (it.hasNext()) {
+                Throwable throwable = it.next();
+                if (throwable instanceof ContentNotSupportedException) {
+                    updateNoVideoDisplay();
+                    it.remove();
+                }
+            }
```

Comment:
Can't you just use `for (Throwable throwable : errors)`?

Reply:
Is it `ConcurrentModificationException` free? I was told that I should use iterator if I want to remove or add things while iterating because it's always safe.

Reply:
@B0pol: You're right. <sup>I'm used to Python.</sup> You should still remove the outer `if` though, and I still prefer making a copy of the list, though that uses more memory. You're also currently assuming there could only be one `ContentNotSupportedException`, while you don't when just making a copy of the list.

Reply:
I'd keep the `if`, but also agree to use a copy of the list to keep the original order.